huaweicloud/services/live/resource_huaweicloud_live_bucket_authorize.go

```diff
@@ -0,0 +1,137 @@
+// ---------------------------------------------------------------
+// *** AUTO GENERATED CODE ***
+// @Product Live
+// ---------------------------------------------------------------
+
+package live
+
+import (
+	"context"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func ResourceLiveBucketAuthorize() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceLiveBucketAuthorizeCreate,
+		ReadContext:   resourceLiveBucketAuthorizeRead,
+		DeleteContext: resourceLiveBucketAuthorizeDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"bucket_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the bucket name of the OBS.`,
+			},
+		},
+	}
+}
+
+func resourceLiveBucketAuthorizeCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	// createLiveBucketAuthorize: create Live snapshot
+	var (
+		createLiveBucketAuthorizeHttpUrl = "v1/{project_id}/obs/authority"
+		createLiveBucketAuthorizeProduct = "live"
+	)
+	createLiveBucketAuthorizeClient, err := cfg.NewServiceClient(createLiveBucketAuthorizeProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating Live Client: %s", err)
+	}
+
+	createLiveBucketAuthorizePath := createLiveBucketAuthorizeClient.Endpoint + createLiveBucketAuthorizeHttpUrl
+	createLiveBucketAuthorizePath = strings.ReplaceAll(createLiveBucketAuthorizePath, "{project_id}",
+		createLiveBucketAuthorizeClient.ProjectID)
+
+	createLiveBucketAuthorizeOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	}
+	createLiveBucketAuthorizeOpt.JSONBody = utils.RemoveNil(buildLiveBucketAuthorizeBodyParams(d, 1))
+	_, err = createLiveBucketAuthorizeClient.Request("PUT", createLiveBucketAuthorizePath,
+		&createLiveBucketAuthorizeOpt)
+	if err != nil {
+		return diag.Errorf("error creating Live bucket authorize: %s", err)
+	}
+
+	d.SetId(d.Get("bucket_name").(string))
+
+	return resourceLiveBucketAuthorizeRead(ctx, d, meta)
+}
+
+func buildLiveBucketAuthorizeBodyParams(d *schema.ResourceData, operation int) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"bucket":    utils.ValueIngoreEmpty(d.Get("bucket_name")),
```

Comment:
```suggestion
		"bucket":    utils.ValueIngoreEmpty(d.Get("bucket")),
```
use bucket is OK, no need to rename

Reply:
done